### PR TITLE
Feature/update clang format 16

### PR DESCRIPTION
**Pull request by godexsoft**
Test clang-format-16 on CI

### CODE REVIEW

File: src/etl/impl/AsyncData.h [hunk 1]
```diff
@@ -94,8 +95,8 @@ class AsyncCallData
         }
         if (!next_->is_unlimited())
         {
-            LOG(log_.warn()) << "AsyncCallData is_unlimited is false. Make sure "
-                                "secure_gateway is set correctly at the ETL source";
+            LOG(log_.warn()
```
Comment (godexsoft) [inline src/etl/impl/AsyncData.h]: Hm, what happened here?

File: src/data/cassandra/impl/Cluster.cpp [hunk 2]
```diff
@@ -50,14 +50,15 @@ Cluster::Cluster(Settings const& settings) : ManagedObject{cass_cluster_new(), c
     cass_cluster_set_token_aware_routing(*this, cass_true);
     if (auto const rc = cass_cluster_set_protocol_version(*this, CASS_PROTOCOL_VERSION_V4); rc != CASS_OK)
```
Comment (kuznetsss) [inline src/data/cassandra/impl/Cluster.cpp]: Don't we want to to put `{` at the same line for loops and ifs. It is controlled by the option:
https://clang.llvm.org/docs/ClangFormatStyleOptions.html#breakbeforebraces
I like `BS_linux` and `BS_WebKit`. WDYT?

File: src/data/cassandra/impl/Cluster.cpp [hunk 3]
```diff
@@ -50,14 +50,15 @@ Cluster::Cluster(Settings const& settings) : ManagedObject{cass_cluster_new(), c
...
     {
-        throw std::runtime_error(
-            fmt::format("Error setting cassandra protocol version to v4: {}", cass_error_desc(rc)));
+        throw std::runtime_error(fmt::format("Error setting cassandra protocol version to v4: {}", cass_error_desc(rc))
```
Comment (kuznetsss) [inline src/data/cassandra/impl/Cluster.cpp:51]: That's a bit weird formatting. I don't know whether it could be fixed.

**Review by kuznetsss — COMMENTED** [review]
Code looks really good, just a few questions

File: src/data/cassandra/impl/Cluster.cpp (hunk 3 above)
Comment (godexsoft) [reply src/data/cassandra/impl/Cluster.cpp:51]: We could set a higher line length limit but i'm not sure it's worth it.

**Review by cindyyan317 — APPROVED** [review]

File: src/data/cassandra/impl/Cluster.cpp (hunk 3 above)
Comment (kuznetsss) [reply src/data/cassandra/impl/Cluster.cpp:51]: It is a bug of clang-format
llvm/llvm-project#55473

**Review by kuznetsss — APPROVED** [review]

**Pull request merged**